packages/synapse-react-client/src/components/EntityViewScopeEditor/EntityViewScopeEditorModal.test.tsx

```diff
@@ -0,0 +1,235 @@
+}))
+const mockEntityViewScopeEditor = jest.mocked(EntityViewScopeEditor)
+
+function renderComponent(props: EntityViewScopeEditorModalProps) {
+  return render(<EntityViewScopeEditorModal {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+const mockUpdateEntity = jest.spyOn(SynapseClient, 'updateEntity')
+
+describe('EntityViewScopeEditorModal tests', () => {
+  const mockOnCancel = jest.fn()
+  const mockOnUpdate = jest.fn()
+
+  beforeEach(() => {
+    jest.spyOn(SynapseClient, 'getEntity').mockResolvedValue({
+      ...mockTableEntity,
+      scopeIds: ['syn123'],
+      viewTypeMask: ENTITY_VIEW_TYPE_MASK_FILE | ENTITY_VIEW_TYPE_MASK_FOLDER,
+      concreteType: ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+    } as Entity)
+  })
+
+  it('displays the correct scope editor and mask editor after initial render', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    expect(
+      within(dialog).getByRole('heading', { name: 'Edit Scope' }),
+    ).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Cancel' })).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
+
+    const filesButton = within(dialog).getByRole('checkbox', { name: 'Files' })
+    const foldersButton = within(dialog).getByRole('checkbox', {
+      name: 'Folders',
+    })
+    const tablesButton = within(dialog).getByRole('checkbox', {
+      name: 'Tables',
+    })
+    const datasetsButton = within(dialog).getByRole('checkbox', {
+      name: 'Datasets',
+    })
+
+    const checkboxes = screen.getAllByRole('checkbox')
+    expect(checkboxes).toHaveLength(4)
+
+    expect(filesButton).toBeVisible()
+    expect(foldersButton).toBeVisible()
+    expect(tablesButton).toBeVisible()
+    expect(datasetsButton).toBeVisible()
+
+    await waitFor(() => {
+      expect(filesButton).toBeChecked()
+      expect(foldersButton).toBeChecked()
+      expect(tablesButton).not.toBeChecked()
+      expect(datasetsButton).not.toBeChecked()
+    })
+  })
+
+  it('display error for invalid entity', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    mockUpdateEntity.mockRejectedValue(
+      new SynapseClientError(
+        404,
+        'Error with scope',
+        expect.getState().currentTestName!,
+      ),
+    )
+
+    const saveButton = within(dialog).getByRole('button', { name: 'Save' })
+    await waitFor(() => {
+      expect(
+        within(dialog).getByRole('button', { name: 'Save' }),
+      ).not.toBeDisabled()
+    })
+    await userEvent.click(saveButton)
+    expect(mockOnUpdate).not.toHaveBeenCalled()
```

Comment:
I would expect that you would waitFor an alert to appear here, verify that the error message is shown, then I would expect mockOnUpdate to have been called (since that's what threw the error)--without that, I am not sure what you are testing here